python/dgl/graph_index.py

```diff
@@ -30,7 +30,8 @@ def __init__(self, handle=None, multigraph=None, readonly=None):
 
     def __del__(self):
         """Free this graph index object."""
-        _CAPI_DGLGraphFree(self._handle)
+        if hasattr(self, '_handle'):
+            _CAPI_DGLGraphFree(self._handle)
```

Comment:
When will '_handle' not exist?

Reply:
When the graph fails to load from the pickle.